pkg/serviceprovider/github/queries.go

```diff
@@ -16,123 +16,62 @@ package github
 
 import (
 	"context"
-	"strconv"
-
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
 
+	"github.com/google/go-github/v45/github"
 	sperrors "github.com/redhat-appstudio/service-provider-integration-operator/pkg/errors"
-	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/httptransport"
-
-	"github.com/machinebox/graphql"
-)
-
-const (
-	allAccessibleAffiliationsReposQuery = `
-		query($after: String) {
-			viewer {
-				repositories(first: 100, after: $after, affiliations: [OWNER, COLLABORATOR, ORGANIZATION_MEMBER]) {
-					pageInfo {
-						hasNextPage
-						endCursor
-					}
-					nodes {
-						viewerPermission
-						url
-					}
-				}
-			}
-		}`
-	allAccessibleOwnerAffiliationsReposQuery = `
-		query($after: String) {
-			viewer {
-				repositories(first: 100, after: $after, ownerAffiliations: [OWNER, COLLABORATOR, ORGANIZATION_MEMBER]) {
-					pageInfo {
-						hasNextPage
-						endCursor
-					}
-					nodes {
-						viewerPermission
-						url
-					}
-				}
-			}
-		}`
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/logs"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 // AllAccessibleRepos lists all the repositories accessible by the current user
 type AllAccessibleRepos struct {
 	Viewer struct {
 		Repositories struct {
-			PageInfo PageInfo `json:"pageInfo"`
-			Nodes    []struct {
+			Nodes []struct {
 				ViewerPermission string `json:"viewerPermission"`
 				Url              string `json:"url"`
 			} `json:"nodes"`
 		} `json:"repositories"`
 	} `json:"viewer"`
 }
 
-func (r *AllAccessibleRepos) FetchAll(ctx context.Context, client *graphql.Client, accessToken string, state *TokenState) error {
+func (r *AllAccessibleRepos) FetchAll(ctx context.Context, githubClient *github.Client, accessToken string, state *TokenState) error {
+
+	lg := log.FromContext(ctx)
+	defer logs.TimeTrack(lg, time.Now(), "fetch all github repositories")
```

Comment:
does timing here works as expected? Is `time.Now()` really evaluated at the beginning of the method? If yes, is it guaranteed by go specification?

Reply:
Yeh. It works. I've checked. Inspiration comes from https://coderwall.com/p/cp5fya/measuring-execution-time-in-go

Reply:
Cool. I see it's in spec "Each time a "defer" statement executes, the function value and parameters to the call are [evaluated as usual](https://go.dev/ref/spec#Calls) and saved anew but the actual function is not invoked." https://go.dev/ref/spec#Defer_statements so it's safe to use.